include/dump_utils.hpp

```diff
@@ -0,0 +1,115 @@
+#pragma once
+
+#include "async_resp.hpp"
+#include "error_messages.hpp"
+#include "logging.hpp"
+#include "utility.hpp"
+
+#include <boost/algorithm/string/case_conv.hpp>
+#include <boost/system/error_code.hpp>
+#include <dbus_singleton.hpp>
+#include <dbus_utility.hpp>
+
+#include <cstddef>
+#include <string>
+
+namespace redfish
+{
+namespace dump_utils
+{
+
+inline void getValidDumpEntryForAttachment(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp, const std::string& url,
+    std::function<void(const std::string& objectPath,
+                       const std::string& entryID,
+                       const std::string& dumpType)>&& callback)
+{
+    std::string dumpType;
+    std::string entryID;
+    std::string entriesPath;
+
+    if (crow::utility::readUrlSegments(url, "redfish", "v1", "Managers", "bmc",
+                                       "LogServices", "Dump", "Entries",
+                                       std::ref(entryID), "attachment"))
+    {
+        // BMC type dump
+        dumpType = "BMC";
+        entriesPath = "/redfish/v1/Managers/bmc/LogServices/Dump/Entries/";
+    }
+    else if (crow::utility::readUrlSegments(
+                 url, "redfish", "v1", "Systems", "system", "LogServices",
+                 "Dump", "Entries", std::ref(entryID), "attachment"))
+    {
+        // System type
+        dumpType = "System";
+        entriesPath = "/redfish/v1/Systems/system/LogServices/Dump/Entries/";
+    }
+    if (dumpType.empty() || entryID.empty())
+    {
+        redfish::messages::resourceNotFound(asyncResp->res, "BMC dump",
```

Comment:
This error should be generic right? not "BMC dumps"

Probably "Dump" or "LogEntry"?

Reply:
Changed to "Dump".

Reply:
But i dont see the change here, i still see "BMC dump"

Reply:
Sorry, I mistakenly did not add the updated file.
It should be changed now.